File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.py
def isRangeCovered(visited, minVal, maxVal):
    i = minVal
    while(i <= maxVal):
        if(visited[i] == False): return False
        i = i + 1
    return True

class Solution:


    def maxChunksToSorted(self, arr: List[int]) -> int:
        n = len(arr)
        visited = [False]*n

        minVal = 0
        maxVal = -10**9 - 1
        l = 0

        count = 0

        while(l < n):
            val = arr[l]
            
            visited[val] = True

            maxVal = max(val, maxVal)

            if(isRangeCovered(visited, minVal, maxVal)):
                count = count + 1
                minVal = l + 1
                maxVal = -10**9 - 1
                l = l + 1
            else:
                l = l + 1

        return count

File: 0780-max-chunks-to-make-sorted/0780-max-chunks-to-make-sorted.cpp
class Solution {
public:
    int maxChunksToSorted(vector<int>& arr) {
        int n = arr.size();
        int maxTillNow = INT_MIN;

        int l = 0, r = 0;

        int count = 0;

        while(l < n){
            int val = arr[l];
            maxTillNow = max(val, maxTillNow);

            // So, if we got max = 2, we must at least go till index 2 for sure.
            if(maxTillNow == l){
                count++;
                maxTillNow = INT_MIN;
                l = l + 1;
            }else{
                l = l + 1;
            }
        }

        return count;
    }
};